src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactoryProvider.java

```diff
@@ -260,6 +265,11 @@ private static boolean convertToBoolean(Object value) {
             builder.port(port);
         }
 
+        Integer fetchSize = connectionFactoryOptions.getValue(FETCH_SIZE);
```

Comment:
Values can be either Integer or String as the URL parser cannot know the actual data type.